src/main/java/tictac7x/charges/items/U_ColossalPouch.java

```diff
@@ -0,0 +1,132 @@
+                new StorageItem(ItemID.PURE_ESSENCE).checkName("Pure essence"),
+                new StorageItem(ItemID.DAEYALT_ESSENCE).checkName("Daeyalt essence"),
+                new StorageItem(ItemID.GUARDIAN_ESSENCE).checkName("Guardian essence")
+            )
+            .maximumTotalQuantity(40);
+
+        this.items = new TriggerItem[]{
+            new TriggerItem(ItemID.COLOSSAL_POUCH),
+            new TriggerItem(ItemID.COLOSSAL_POUCH_26786), // Degrated
+        };
+
+        this.triggers = new TriggerBase[]{
+            // Empty
+            new OnChatMessage("There is no essence in this pouch.").emptyStorage(),
+            new OnChatMessage("The rift becomes active!").emptyStorage(),
+
+            // Check
+            new OnChatMessage("There (?:is|are) (?<charges>.+?) (?<essence>pure|daeyalt|guardian)? ?essences? in this pouch.").matcherConsumer((m) -> {
+                storage.clear();
+
+                int itemID;
+                switch (m.group("essence")) {
+                    case "pure":
+                        itemID = ItemID.PURE_ESSENCE;
+                        break;
+                    case "daeyalt":
+                        itemID = ItemID.DAEYALT_ESSENCE;
+                        break;
+                    case "guardian":
+                        itemID = ItemID.GUARDIAN_ESSENCE;
+                        break;
+                    default:
+                        itemID = ItemID.RUNE_ESSENCE;
+                        break;
+                }
+                
+                storage.put(itemID, getNumberFromWordRepresentation((m.group("charges"))));
+            }).onMenuOption("Check"),
+
+            // Decay.
+            new OnChatMessage("Your pouch has decayed through use.").onMenuOption("Fill").consumer(() -> {
+                this.decayCount += 1;
+                this.storage.maximumTotalQuantity(getMaxCharges());
+            }),
+            
+            // Repair.
+            new OnChatMessage( "Fine. A simple transfiguration spell should resolve things for you.").consumer(() -> {
+                this.decayCount = 0;
+                this.storage.maximumTotalQuantity(getMaxCharges());
+            }),
+
+            // Fill from inventory.
+            new OnItemContainerChanged(ItemContainerId.INVENTORY).fillStorageFromInventorySingle().onMenuOption("Fill"),
+
+            // Fill from bank.
+            new OnItemContainerChanged(ItemContainerId.BANK).fillStorageFromInventorySingle().onMenuOption("Fill"),
+
+            // Use essence on pouch.
+            new OnItemContainerChanged(INVENTORY).fillStorageFromInventorySingle().onUseStorageItemOnChargedItem(storage.getStoreableItems()),
+
+            // Empty to inventory.
+            new OnItemContainerChanged(ItemContainerId.INVENTORY).emptyStorageToInventory().onMenuOption("Empty"),
+
+            // Set maximum charges on level up
+            new OnStatChanged(Skill.RUNECRAFT).consumer(() -> {
```

Comment:
I couldn't use `OnXpDrop` as that actually checks for XP change. I need this to also work "on login" when XP is initially set.

Reply:
base on https://github.com/TicTac7x/runelite-plugins/tree/plugin-charges-v0.5.12

Reply:
Unfortunately this doesn't work. It's not the consumer that's the problem, it's the checks that the XpDrop listener does which cause it not to run when we first login. I need that first login XP set to initialize the pouch, _then_ after any level ups too.

Reply:
Oh yeah, the other issue I found was the event itself has the `inInventoryOrEquipment` check before triggering. But I need the state change regardless of what we're looking at.